fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -30,6 +30,7 @@ import org.mozilla.fenix.wallpapers.Wallpaper
  * [Action] implementation related to [AppStore].
  */
 sealed class AppAction : Action {
+    data class ChangeScreen(val screen: Screen) : AppAction()
```

Comment:
I might be getting hung up on semantics, but what about:

```
data class Navigate(val destination: NavigationDestination) : AppAction()
```

There will be instances, like Tabs Tray or the review checker, where we navigate to a non-fullscreen feature flow.

Tangentially, but I imagine at some point our CFRs getting an overhaul and having an engine that's centered around a similar Redux pattern.